coc/http.py

```diff
@@ -67,71 +68,62 @@ def __init__(self, method, path, kwargs: dict=None, api_page=False):
 
 
 class HTTPClient:
-    def __init__(self, client, loop, tokens=None, *, email, password, update_tokens=False):
+    def __init__(self, client, loop, email, password,
+                 key_names, key_count):
         self.client = client
         self.loop = loop
-        self.__session = None
+        self.__session = aiohttp.ClientSession(loop=self.loop)
```

Comment:
The reason this was None and set on `login()` was that aiohttp throws annoying unclosed client session if its made on init. I'm happy to test this again, but suspect same results

Reply:
I tested, it works fine. I believe I fixed it. The issue is not closing it so just need to make sure if something we do forces the client to close unexpectedly then we close it before it closes itself.

P.S. For clarification It only throws that warning when the session is closed without an `await session.close()` Discord.py can throw that too if you do something wrong too just fyi

P.S.S. If you're using this Client in a Discord cog and define the Client in `__init__` then override the `__unload` method and use `await Client.close()` in there.

Reply:
Ok sweet. I always had trouble in the past and it must've been a lucky coincidence it worked the way I did it before.
Yep, but I believe its been renamed to `on_cog_unload` ;)

Reply:
Haha so many changes lol